### DISCUSSION TITLE
Run in background on Android

### DISCUSSION THREAD

**Discussion by Ahmetozefe** [localsend/localsend#141] [Discussion (English, International)]
Right now, when I want to send a file to my phone, I have to open LocalSend on the phone otherwise the PC won't see it. Is it possible to have it run in the background so it's always discoverable?

**Comment by Yuslendo** [comment]
my thoughts are that running the app all the time in the background will consume battery. so I think this feature is probably not will added. you can actually leave the app open in the background when sending or receiving starts, but android can maybe turn off LocalSend to speed up the device and make the resource go to the active app.

**Reply by Arnab-Afk** [reply]
I think running the whole app in background would be not a great idea instead just it should have permissions to keep connected whole the time so that it would also save the battery and whole application won't be running all the time.

**Reply by GittBittu** [reply]
implementing it like android's nearby share could be a great idea. just give a toggle in the quick settings panel and turn on or off the app through there.

**Comment by trymeouteh** [comment]
Would be a nice option to have

**Comment by rpbp58** [comment]
The only thing that would make this unique or better over nearby share would be if it ran in the background on my android phone.  Having to take out my phone and open the app is just as clunky and bothersome as nearby share.

The ultimate scenario would be if I could make my phone appear as a wifi network drive on Windows.  Since that doesn't seem to be possible, the next most awesome scenario would be to right-click a file in Explorer and send-to my phone, without having to touch my phone.

**Comment by jbender0** [comment]
Glad to know I'm not the only one thinking this!

**Comment by thewaliii** [comment]
At least give us the option, and disable it by default. I don't care about the battery.

**Comment by Koboku** [comment]
Please add this option, and if you wish, disable it by default.

 I think that sending a file should take minimal amount of actions. So it's why we have option to choose save folder and favorite devices list, but it can be even quicker:
- No need to open a mobile app, it runs in the background 
- Send from PC context menu
- Send Directly to one default device, without choosing it from the list (additional context menu option)

**Comment by soloam** [comment]
I think that a option to do this would be a must have... Please consider this as a optional feaure

**Comment by Starship-Headlight** [comment]
+1 for this. Really needed. It gets really frustrating to cancel the failed transfer and start the transfer again. Please consider adding this 🙏

**Comment by tsuki-xi77** [comment]
+1 here

**Comment by Rasit-Vatan** [comment]
Yes please...

**Comment by MO77-LXXVII** [comment]
Will be very helpful since moving large files from pc to phone fail sometimes, as the phone screen timeout automatically.

**Comment by AoSankaku** [comment]
When this feature will be added I really need this

**Comment by d-cdmi** [comment]
+1

**Comment by HamonTeacher** [comment]
The need to keep an app open while transferring files seems wild. A messenger notification came up and you switched there - back there? That's it, transmission error. It's really annoying.

**Comment by johnsageeek** [comment]
Sometimes, turning off "Battery Optimization" for an app keep's it o[pen. However, won't let you run it in the background.

**Comment by LetsCNow** [comment]
Me too. Bluetooth is always listening, why can't the localsend receiver do also?

Absolutely fabulous software though.

**Comment by I21b** [comment]
New duplicate #989

**Comment by HappySnowman821** [comment]
Also would really ask for this feature. I debloat my devices and nowadays they are conditioned to so much bloat and bs anyways, battery isn't a concern. 
And for security / privacy - maybe add the option to ask for a token to even consider a request Ig
Or add the option to require the device to be awake 

Even if it takes ADB permissions or root to work without compromises, this would make this project a full blown competitor to quick share or honestly the only real option for WiFi direct sharing imo.

This would save so much time when in class or when transferring documents to my tablet at home.

**Comment by aarron-lee** [comment]
Just a thought on implementation, syncthing-fork is able to run in the background on Android.

I presume it's because Syncthing on Android has a persistent notification, and Syncthing turns itself off when no wifi is connected.

Could something similar be done on with Localsend? Since Localsend requires a wifi connection anyways, this would fit perfectly.

Ideally, a notification would prompt when a file is being sent, and you could additionally implement something like "auto-accept from trusted devices"

## Syncthing With Wifi

<img width="400" alt="with_wifi" src="https://github.com/user-attachments/assets/6021d63e-7f5a-4f8c-a0c3-b36f3590cd77" />

## Syncthing Without Wifi

<img width="400" alt="without_wifi" src="https://github.com/user-attachments/assets/5a035557-c6ca-4ce1-9614-1901ab71969e" />
